cclient/response/were_addresses_spent_from.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "were_addresses_spent_from.h"
+
+were_addresses_spent_from_res_t* were_addresses_spent_from_res_new() {
+  were_addresses_spent_from_res_t* states = NULL;
+  utarray_new(states, &ut_int_icd);
+  return states;
+}
+
+void were_addresses_spent_from_res_free(were_addresses_spent_from_res_t* res) {
+  utarray_free(res);
+}
+
+bool were_addresses_spent_from_res_at(were_addresses_spent_from_res_t* in,
+                                      int index) {
+  if (utarray_len(in) > index) {
+    bool* p = NULL;
+    p = (bool*)utarray_eltptr(in, index);
+    if (p) {
+      return *p;
+    }
+  }
+  return NULL;
```

Comment:
can we change this function to return `retcode_t` and accept `bool*` ?

Reply:
I found it's kind strange to return `retcode_t` intstead of `bool`, since the similar function in the other API returns `bool`.
Should I change or all the similar functions in other APIs, or just leaves it in the original way?
thank you

Reply:
hey @HowJMay ,
It is because you return NULL as boolean that i say so, otherwise i agree we could not return `retcode`
so any place that has NULL returned s a boolean should return RC_CCLIENT_OUT_OF_BOUND_INDEX or something like that..
but for this PR this place is enough :)
Thanks